src/lm_buddy/jobs/common.py

```diff
@@ -0,0 +1,49 @@
+import contextlib
+import tempfile
+from enum import Enum
+from pathlib import Path
+
+from pydantic_yaml import parse_yaml_file_as, to_yaml_file
+
+from lm_buddy.types import BaseLMBuddyConfig
+
+
+class LMBuddyJobType(str, Enum):
+    """Enumeration of logical job types runnable via the LM Buddy."""
+
+    PREPROCESSING = "preprocessing"
+    FINETUNING = "finetuning"
+    EVALUATION = "evaluation"
+
+
+class LMBuddyJobConfig(BaseLMBuddyConfig):
+    """Configuration that comprises the entire input to an LM Buddy job.
+
+    This class implements helper methods for de/serializing the configuration from file.
+
+    Currently, there is a 1:1 mapping between job entrypoints and job config implementations,
+    but this is not rigidly constrained by the interface. This may change in the future.
+    """
+
+    @classmethod
```

Comment:
I've intentionally moved the YAML file methods onto this subclass to restrict the interface.

Currently, the only supported case we have for writing to/from file is for full job configs, not the independent sub-components. So it feels a bit more natural to constrain that functionality to this subclass rather than on the parent `BaseLMBuddyConfig` class.